### PR TITLE
prepare for push notifications #6707

### LINKED ISSUE #6707: Implement push notification
Refer https://github.com/AlphaWallet/alpha-wallet-android/issues/3152

Should need some details like our backend's web API endpoints for registering a wallet and the payload format (suggest changes if you need, to support both iOS and Android). @justindg should have them

### PR DESCRIPTION
**Pull request by oa-s**
Closes #6707

### CODE REVIEW

File: modules/AlphaWalletNotifications/.gitignore [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Xcode
```
Comment (hboon) [inline modules/AlphaWalletNotifications/.gitignore:1]: I think we don't need the `.gitignore` file right? Maybe a remnant from when we (wanted?) to have a separate repo for each? Shall we remove them separately?

```
./modules/AlphaWalletCore/.gitignore
./modules/AlphaWalletTrackAPICalls/.gitignore
./modules/AlphaWalletAddress/.gitignore
./modules/AlphaWalletLogger/.gitignore
```

File: modules/AlphaWalletNotifications/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
+
+Copyright (c) 2020 AlphaWallet
```
Comment (hboon) [inline modules/AlphaWalletNotifications/LICENSE:3]: 2023 :)

File: AlphaWalletTests/ViewControllers/PaymentCoordinatorTests.swift [hunk 3]
```diff
@@ -24,10 +24,14 @@ extension CurrencyService {
 }
 
 extension WalletDataProcessingPipeline {
+<<<<<<< HEAD
```
Comment (hboon) [inline AlphaWalletTests/ViewControllers/PaymentCoordinatorTests.swift:27]: There are a few merge conflict markers

**Review by hboon — COMMENTED** [review]
Also problem building after removing the merge conflict markers

**Comment by hboon** [comment]
Seems to include unrelated changes like for `SwitchTableViewCell`?

**Comment by oa-s** [comment]
for this pr `SwitchTableViewCell` was needed to update with adding loadable state, for treaking push notification state enable/disable

File: modules/AlphaWalletNotifications/.gitignore (hunk 1 above)
Comment (oa-s) [reply modules/AlphaWalletNotifications/.gitignore:1]: don't know, will remove `.gitignore` related to this pr, the rest in another pr

**Comment by hboon** [comment]
I'm getting this build error:

> (103, 25) Value of type 'Loadable<TransactionConfirmationViewModel.SendFungiblesTransactionViewModel.TransactedToken, any Error>' has no member 'mapValue'

Just me?

**Comment by hboon** [comment]
Does it actually show a local notification already? If it does, can we hide a feature flag to disable it?

**Comment by oa-s** [comment]
Yes, it does

**Comment by hboon** [comment]
The feature flag should disable the registration and the accepting the notification and starting processing, so something like:

```
if feature_flag
    prompt_user_to_get_device_token
end

when_push_received
    if feature_flag
        refresh_timelines
        show_local_notification
    end
end
```

**Comment by oa-s** [comment]
> mapValue

will check

**Comment by oa-s** [comment]
@hboon  our current implementation of fetching transactions suppose to make timed call of transaction request (it looks like a bit different from Androids impl), i checked Android implementation and maybe not clearly understood but in general they just fetch latest transactions once a push notification received. Would it be correctly for us just run timers that make appropriate requests?

our SingleChainTransactionProvider has `start` method, so we could stop service in case its in `running` state, e.g waiting for next cycle call. and force start timer again. Our `TransactionsNotificationSource` will handle received transactions and will create appropriate local notifications.

**Comment by oa-s** [comment]
> So upon receiving the remote notification, we just do this? —
> ``` _pull_with_etherscan() ```

yes

> This is what Android does too right?

yes

**Comment by oa-s** [comment]
>  pull_transactions()

what does it should do?

**Comment by oa-s** [comment]
> The push notification just triggers it right away.

Ok i see. No u understood it right, that is just me don't understood right what push notification should do, but now i do. So yes we can trigger to pull transactions by push notification (meaning to restart fetch transactions timers).

**Comment by oa-s** [comment]
firstly i thought that all needed data will be in the received push notification, so we wouldn't need to do any additional actions to display push notification, but now i got it

**Comment by hboon** [comment]
Yes. The remote notification is like a refresh button :) it doesn’t contain much information. For context, it can’t contain much information because we don’t have a backend wallet API that the wallet syncs with.

**Comment by oa-s** [comment]
@hboon it looks like we are using different projects for ios and android applications, could u please check android app if u have access to it and invite me to it. Need choose one from two. push notification api configured with android project

**Comment by hboon** [comment]
@oa-s seems like the server has now moved to using the project iOS is on right? Let's coordinate with Justin and move them to use the iOS one?

**Pull request merged**